diff --git a/sip/DockManager.sip b/sip/DockManager.sip
index b76ee25..941fca7 100644
--- a/sip/DockManager.sip
+++ b/sip/DockManager.sip
@@ -135,7 +135,6 @@ protected:
 	ads::CDockOverlay* dockAreaOverlay() const;
     void notifyWidgetOrAreaRelocation(QWidget* RelocatedWidget);
     void notifyFloatingWidgetDrop(ads::CFloatingDockContainer* FloatingWidget);
-    ads::CDockWidget* focusedDockWidget() const;
 
     virtual void showEvent(QShowEvent *event);
 
@@ -217,6 +216,9 @@ public:
 	void setViewMenuInsertionOrder(ads::CDockManager::eViewMenuInsertionOrder Order);
 	bool isRestoringState() const;
 	static int startDragDistance();
+    ads::CDockWidget* focusedDockWidget() const;
+    QList<int> splitterSizes(ads::CDockAreaWidget *ContainedArea) const;
+    void setSplitterSizes(ads::CDockAreaWidget *ContainedArea, const QList<int>& sizes);
 
 public slots:
 	void openPerspective(const QString& PerspectiveName);
diff --git a/sip/DockWidgetTab.sip b/sip/DockWidgetTab.sip
index 32d819b..0bc610a 100644
--- a/sip/DockWidgetTab.sip
+++ b/sip/DockWidgetTab.sip
@@ -35,6 +35,8 @@ public:
 	virtual bool event(QEvent *e);
 	void setElideMode(Qt::TextElideMode mode);
     void updateStyle();
+    QSize iconSize() const;
+    void setIconSize(const QSize& Size);
 
 public slots:
 	virtual void setVisible(bool visible);
