diff --git a/setup.py b/setup.py
index 6ae6e74..3d861b0 100644
--- a/setup.py
+++ b/setup.py
@@ -1,6 +1,7 @@
 import os
 import sys
 import shlex
+import shutil
 import subprocess
 import glob
 
@@ -140,8 +141,23 @@ class build_ext(sipdistutils.build_ext):
                              'Please specify via --pyqt-sip-flags=')
 
     def _find_sip(self):
-        """override _find_sip to allow for manually speficied sip path."""
-        return self.sip_bin or super()._find_sip()
+        """override _find_sip to allow for manually-specified sip path."""
+        # 1. Manually specified sip_bin
+        if self.sip_bin:
+            return self.sip_bin
+
+        # 2. Path determined from sipconfig.Configuration()
+        #    This may not exist, depending on conda build configuration.
+        sip_bin = super()._find_sip()
+        if os.path.exists(sip_bin):
+            return sip_bin
+
+        # 3. Finally, fall back to sip binary found in path
+        sip_bin = shutil.which('sip')
+        if sip_bin:
+            return sip_bin
+
+        raise SystemExit('Could not find PyQt SIP binary.')
 
     def _sip_compile(self, sip_bin, source, sbf):
         cmd = [sip_bin]
diff --git a/sip/DockManager.sip b/sip/DockManager.sip
index 19873e6..7a6d5e7 100644
--- a/sip/DockManager.sip
+++ b/sip/DockManager.sip
@@ -165,12 +165,12 @@ public:
 		DockAreaDynamicTabsMenuButtonVisibility,
 		FloatingContainerHasWidgetTitle,
 		FloatingContainerHasWidgetIcon,
+		HideSingleCentralWidgetTitleBar,
         DefaultDockAreaButtons,
 		DefaultBaseConfig,
         DefaultOpaqueConfig,
 		DefaultNonOpaqueConfig,
 		NonOpaqueWithWindowFrame,
-		HideSingleCentralWidgetTitleBar,
 	};
     typedef QFlags<ads::CDockManager::eConfigFlag> ConfigFlags;
 
diff --git a/sip/DockWidgetTab.sip b/sip/DockWidgetTab.sip
index f6c2453..d32a1d2 100644
--- a/sip/DockWidgetTab.sip
+++ b/sip/DockWidgetTab.sip
@@ -33,7 +33,7 @@ public:
 	bool isTitleElided() const;
 	bool isClosable() const;
 	virtual bool event(QEvent *e);
-
+	void setElideMode(Qt::TextElideMode mode);
 
 public slots:
 	virtual void setVisible(bool visible);
diff --git a/sip/ads_globals.sip b/sip/ads_globals.sip
index 66b0dbe..ea78457 100644
--- a/sip/ads_globals.sip
+++ b/sip/ads_globals.sip
@@ -10,7 +10,7 @@ namespace ads
 
     enum eStateFileVersion
     {
-        InitialVerison,
+        InitialVersion,
         Version1,
         CurrentVersion
     };
