diff --git a/sip/DockContainerWidget.sip b/sip/DockContainerWidget.sip
index 8c5002e..ae90e3c 100644
--- a/sip/DockContainerWidget.sip
+++ b/sip/DockContainerWidget.sip
@@ -23,7 +23,7 @@ protected:
 	QSplitter* rootSplitter() const;
 	void createRootSplitter();
 	void dropFloatingWidget(ads::CFloatingDockContainer* FloatingWidget, const QPoint& TargetPos);
-    void dropWidget(QWidget* widget, const QPoint& TargetPos);
+    void dropWidget(QWidget* Widget, DockWidgetArea DropArea, CDockAreaWidget* TargetAreaWidget);
 	void addDockArea(ads::CDockAreaWidget* DockAreaWidget /Transfer/, ads::DockWidgetArea area = ads::CenterDockWidgetArea);
 	void removeDockArea(ads::CDockAreaWidget* area /TransferBack/);
 	void saveState(QXmlStreamWriter& Stream) const;
diff --git a/sip/DockOverlay.sip b/sip/DockOverlay.sip
index 040d634..1588800 100644
--- a/sip/DockOverlay.sip
+++ b/sip/DockOverlay.sip
@@ -24,6 +24,7 @@ public:
 	void setAllowedAreas(ads::DockWidgetAreas areas);
 	ads::DockWidgetAreas allowedAreas() const;
 	ads::DockWidgetArea dropAreaUnderCursor() const;
+    ads::DockWidgetArea visibleDropAreaUnderCursor() const;
 	ads::DockWidgetArea showOverlay(QWidget* target);
 	void hideOverlay();
 	void enableDropPreview(bool Enable);
diff --git a/sip/DockSplitter.sip b/sip/DockSplitter.sip
index 82775ef..d6a5f31 100644
--- a/sip/DockSplitter.sip
+++ b/sip/DockSplitter.sip
@@ -16,6 +16,9 @@ public:
 	CDockSplitter(Qt::Orientation orientation, QWidget *parent /TransferThis/ = 0);
 	virtual ~CDockSplitter();
 	bool hasVisibleContent() const;
+	QWidget* firstWidget() const;
+	QWidget* lastWidget() const;
+
 };
 
 };
