diff --git a/setup.py b/setup.py
index 1986df7..98b4063 100644
--- a/setup.py
+++ b/setup.py
@@ -228,6 +228,8 @@ class build_ext(sipdistutils.build_ext):
                     '-mmacosx-version-min=10.9']
             elif sys.platform == 'linux':
                 extension.extra_compile_args += ['-std=c++11']
+                for path in glob.glob(os.path.join(self.qt_include_dir, 'QtGui', '**', 'QtGui', 'qpa'), recursive=True):
+                    extension.include_dirs.append(path)
 
         return super().swig_sources(sources, extension)
 
