diff --git a/project.py b/project.py
index a8233cd..9c79d24 100644
--- a/project.py
+++ b/project.py
@@ -1,4 +1,5 @@
 import os
+import sys
 
 from pyqtbuild import PyQtBindings, PyQtProject
 from sipbuild import Option
@@ -17,11 +18,39 @@ class ads(PyQtBindings):
 
 		super().__init__(project, 'ads')
 
+	def get_options(self):
+		"""Our custom options that a user can pass to sip-build."""
+		options = super().get_options()
+		options += [
+			Option('ads_incdir',
+					help='the directory containing the ads header file',
+					metavar='DIR'),
+			Option('ads_libdir',
+					help='the directory containing the ads library',
+					metavar='DIR'),
+			Option('ads_lib',
+					help='the ads library',
+					metavar='LIB'),
+		]
+		return options
+
 	def apply_user_defaults(self, tool):
 		""" Set default values for user options that haven't been set yet. """
 
 		resource_file = os.path.join(self.project.root_dir,'src','ads.qrc')
 		print("Adding resource file to qmake project: ", resource_file)
 		self.builder_settings.append('RESOURCES += '+resource_file)
+
+		if sys.platform == "linux":
+			self.builder_settings.append('QT += gui-private')
 		
+		if self.ads_lib is not None:
+			self.libraries.append(self.ads_lib)
+
+		if self.ads_incdir is not None:
+			self.include_dirs.append(self.ads_incdir)
+
+		if self.ads_libdir is not None:
+			self.library_dirs.append(self.ads_libdir)
+
 		super().apply_user_defaults(tool)
